runtime/commonMain/src/kotlinx/serialization/internal/SerialClassDescImpl.kt

```diff
@@ -6,24 +6,29 @@ package kotlinx.serialization.internal
 
 import kotlinx.serialization.*
 import kotlinx.serialization.CompositeDecoder.Companion.UNKNOWN_NAME
-import kotlin.jvm.JvmOverloads
+import kotlin.jvm.*
 
-open class SerialClassDescImpl @JvmOverloads constructor(
-    override val name: String,
+/**
+ * Implementation that plugin uses to implement descriptors
+ * for auto-generated serializers.
+ *
+ * Unused methods are invoked by auto-generated plugin code
+ */
+@InternalSerializationApi
+public open class SerialClassDescImpl @JvmOverloads constructor(
```

Comment:
Do we have a migration plan for the users that were using it for custom serializers?

Reply:
Yes we do, a separate PR for that